client/peerset/src/peersstate.rs

```diff
@@ -169,8 +169,8 @@ impl PeersState {
 
 	/// Returns an object that grants access to the reputation value of a peer.
 	pub fn peer_reputation(&mut self, peer_id: PeerId) -> Reputation {
-		if !self.nodes.contains_key(&peer_id) {
-			self.nodes.insert(peer_id, Node::new(self.sets.len()));
+		if let Entry::Vacant(e) = self.nodes.entry(peer_id) {
```

Comment:
```suggestion
		self.nodes.entry(peer_id).or_insert_with(|| Node::new(self.sets.len()))
```

This entire line can replace everything up to the let entry below